core/src/test/scala/unit/kafka/utils/ZkUtilsTest.scala

```diff
@@ -40,6 +40,24 @@ class ZkUtilsTest extends ZooKeeperTestHarness {
     assertTrue("Deletion should be successful", zkUtils.conditionalDeletePath(path, 0))
   }
 
+  /*
+   * Added this test since PIDManager's behavior is depending on the expected return values
```

Comment:
Nit: it seems odd to have "Added this test" as a test comment. Something like "Verify behaviour of ZkUtils.createSequentialPersistentPath since PIDManager relies on it" seems like the expected style.